File: 1304_Find _Unique_Integers_Sum_up_to_Zero.py
class Solution:
    def sumZero(self, n: int) -> list[int]:
        result = []
        for num in range( 1, (n // 2)+1 ):
            result.extend([(-1 * num), num])

        if n % 2 != 0:
            result.append(0)
        
        return result
    

a = Solution()
print(a.sumZero(5))     # [-2, -1, 0, 1, 2]
print(a.sumZero(3))     # [-1, 0, 1]

File: 1304_Find _Unique_Integers_Sum_up_to_Zero.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int* sumZero(int n, int* returnSize) {
    int* result = (int*)malloc(n * sizeof(int));
    int index = 0;

    for (int num = 1; num <= n / 2; num++) {
        result[index++] = -num;
        result[index++] = num;
    }

    if (n % 2 != 0) {
        result[index++] = 0;
    }

    *returnSize = n;
    return result;
    
}

int main() {
    int returnSize;
    int* result = sumZero(5, &returnSize);

    for (int i = 0; i < returnSize; i++) {
        printf("%d ", result[i]);
    }

    printf("\n");
    free(result);
    
    return 0;
}